### DISCUSSION TITLE
Can't Create Functions - [WinError 32] The process cannot access the file because it is being used by another process

### DISCUSSION THREAD

**Discussion by github-pmj** [open-webui/open-webui#5606] [General]
# Bug Report

## Installation Method

.venv 
py -m pip install --upgrade open-webui

## Environment

- **Open WebUI Version:**0.3.23
- **Ollama (if applicable):** 0.3.11

- **Operating System:** Windows 10 22H2
- **Browser (if applicable):** Edge

**Confirmation:**

- [X ] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [X ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [X ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

The function should be saved

## Actual Behavior:

An error is returned and the function is not saved

## Description

**Bug Summary:**
Unable to save a Function from Open-WebUI. After clicking Save, and confirming the action I receive a Win32 error relating to a temporary file being created by the process.:- "[WinError 32] The process cannot access the file because it is being used by another process" 

## Reproduction Details

**Steps to Reproduce:**
Attempt to add the sample filter function - providing a name and description (Test, Test).
Click Save and Confirm Action
Error is received

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]
[Function_Save_Error_Broswer.log](https://github.com/user-attachments/files/17089167/Function_Save_Error_Broswer.log)

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots/Screen Recordings (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]
<img width="400" alt="OWI_Function_Error" src="https://github.com/user-attachments/assets/c03f1690-3536-46f4-ae11-64ed5ad4c466">

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
Unable to reproduce, PR welcome here!

**Comment by iamexemplar** [comment]


**Reply by github-pmj** [reply]
No, hadn't even heard of it to be honest

**Comment by github-pmj** [comment]
I used ProcMon to monitor the temp folder - the only process accessing the file is Python. This is the log that shows the SHARING VIOLATION. (Tried attaching as zip and as txt but kept getting removed)
"Time of Day","Process Name","PID","Operation","Path","Result","Detail"
"16:32:35.6733952","python.exe","8204","CreateFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS","Desired Access: Generic Read/Write, Disposition: Create, Options: Synchronous IO Non-Alert, Non-Directory File, Attributes: N, ShareMode: Read, Write, AllocationSize: 0, OpenResult: Created"
"16:32:35.6735660","python.exe","8204","QueryInformationVolume","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS","VolumeCreationTime: 29 Sep 15:04:51, VolumeSerialNumber: D880-0FCD, SupportsObjects: True, VolumeLabel: "
"16:32:35.6735879","python.exe","8204","QueryAllInformationFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","BUFFER OVERFLOW","CreationTime: 22 Sep 16:32:35, LastAccessTime: 22 Sep 16:32:35, LastWriteTime: 22 Sep 16:32:35, ChangeTime: 22 Sep 16:32:35, FileAttributes: A, AllocationSize: 0, EndOfFile: 0"
"16:32:35.6737527","python.exe","8204","CreateFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS","Desired Access: Generic Write, Read Attributes, Disposition: OverwriteIf, Options: Synchronous IO Non-Alert, Non-Directory File, Attributes: N, ShareMode: Read, Write, AllocationSize: 0, OpenResult: Overwritten"
"16:32:35.6738529","python.exe","8204","QueryInformationVolume","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS","VolumeCreationTime: 29 Sep 15:04:51, VolumeSerialNumber: D880-0FCD, SupportsObjects: True, VolumeLabel: "
"16:32:35.6738796","python.exe","8204","QueryAllInformationFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","BUFFER OVERFLOW","CreationTime: 22 Sep 16:32:35, LastAccessTime: 22 Sep 16:32:35, LastWriteTime: 22 Sep 16:32:35, ChangeTime: 22 Sep 16:32:35, FileAttributes: A, AllocationSize: 0, EndOfFile: 0"
"16:32:35.6739355","python.exe","8204","WriteFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS","Offset: 0, Length: 2,717, Priority: Normal"
"16:32:35.6740398","python.exe","8204","CloseFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS",""
"16:32:35.6763590","python.exe","8204","CreateFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS","Desired Access: Read Attributes, Disposition: Open, Options: Open Reparse Point, Attributes: n/a, ShareMode: Read, Write, Delete, AllocationSize: n/a, OpenResult: Opened"
"16:32:35.6763953","python.exe","8204","QueryNetworkOpenInformationFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS","CreationTime: 22 Sep 16:32:35, LastAccessTime: 22 Sep 16:32:35, LastWriteTime: 22 Sep 16:32:35, ChangeTime: 22 Sep 16:32:35, AllocationSize: 01 Jan 01:00:00, EndOfFile: 01 Jan 01:00:00, FileAttributes: A"
"16:32:35.6764165","python.exe","8204","CloseFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS",""
"16:32:35.6765726","python.exe","8204","CreateFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SHARING VIOLATION","Desired Access: Read Attributes, Delete, Disposition: Open, Options: Non-Directory File, Open Reparse Point, Attributes: n/a, ShareMode: Read, Write, Delete, AllocationSize: n/a"
"16:32:35.6773308","python.exe","8204","CloseFile","<snipped>\AppData\Local\Temp\tmp2c_y3cec","SUCCESS",""

**Comment by tjbck** [comment]
I don't have a Windows machine to be able to reproduce this issue, if anyone could identify the issue and make a fix PR, I'd greatly appreciate it!

**Reply by EtiennePerot** [reply]
Sent #5649 which I believe should fix it (though I also don't have a real Windows machine to test this on)
